application/settings.gradle.kts

```diff
@@ -35,62 +41,135 @@ val coilVersion = "2.1.0"
 val sqliteVersion = "2.2.0"
 val roomVersion = "2.5.1"
 
-dependencyResolutionManagement {
-    versionCatalogs {
-        create("libs") {
-            addLibrary("com.jakewharton.timber:timber:4.7.1")
-            addBundle(
-                name = "coil",
-                "io.coil-kt:coil:$coilVersion",
-                "io.coil-kt:coil-base:$coilVersion"
-            )
-            addBundle(
-                name = "room",
-                "androidx.sqlite:sqlite:$sqliteVersion",
-                "androidx.sqlite:sqlite-framework:$sqliteVersion",
-                "androidx.room:room-runtime:$roomVersion",
-                "androidx.room:room-ktx:$roomVersion",
-                "androidx.room:room-common:$roomVersion",
-            )
-            addPlugin("tools.forma.demo:dependencies", "0.0.1")
-            addPlugin(
-                "com.google.devtools.ksp",
-                "$embeddedKotlinVersion-1.0.9",
-                "androidx.room:room-compiler:$roomVersion"
-            )
+val ksp = CustomConfiguration("ksp")
+
+fun projectDependencies(name: String = "libs", vararg deps: Any) {
+    dependencyResolutionManagement {
+        versionCatalogs {
+            create(name) {
+                deps.forEach {
+                    when (it) {
+                        is String -> addLibrary(it)
+                        is BundleDep -> addBundle(it)
+                    }
+                }
+
+                addPlugin("tools.forma.demo:dependencies", "0.0.1")
+                addPlugin(
+                    id = "com.google.devtools.ksp:symbol-processing-gradle-plugin",
+                    version = "$embeddedKotlinVersion-1.0.10",
+                    configuration = ksp,
+                    "androidx.room:room-compiler:$roomVersion"
+                )
+            }
         }
     }
 }
```

Comment:
why are you avoid toml?)

Reply:
Toml is very limited and does not allow to provide any dependency metadata
Nice bonus of not using toml - now we can generate dependency names based on artifact name, which is important because one of the next features would be - generating list of transitive dependencies